### ISSUE TITLE
ONNX checker doesn't check repeated permutation for Transpose operator

### ISSUE THREAD

**Issue by GuanLuo** [onnx/onnx#3709]
# Bug Report

### Is the issue related to model conversion?
If the ONNX checker reports issues with this model then this is most probably related to the converter used to convert the original framework model to ONNX. Please create this bug in the appropriate converter's GitHub repo (pytorch, tensorflow-onnx, sklearn-onnx, keras-onnx, onnxmltools) to get the best help.

### Describe the bug
The ONNX checker doesn't check if `perm` attribute contains duplicate entry, which is checked in numpy. The sample code below repeats 0 axis and passed the check.

### System information
- OS Platform and Distribution (*e.g. Linux Ubuntu 16.04*): Ubuntu 20.04
- ONNX version (*e.g. 1.7*):  1.10.1
- Python version: 3.8.10

### Reproduction instructions
- Describe the code to reproduce the behavior.
```
from onnx import helper
from onnx import TensorProto as tp
from onnx import checker
from onnx import save
from onnx import shape_inference
inputs = [
    ("input", tp.FLOAT, [2, 3, 4]),
]
weights = []
outputs = [
    ("output", tp.FLOAT, [4, 2, 2]),
]
node = helper.make_node('Transpose',
                        inputs=[input[0] for input in (inputs + weights)],
                        outputs=[output[0] for output in outputs],
                        perm=[2, 0, 0])
graph = helper.make_graph(
    [node], 'test_graph',
    [helper.make_tensor_value_info(*input) for input in inputs],
    [helper.make_tensor_value_info(*output) for output in outputs],
    [helper.make_tensor(*weight) for weight in weights])
model = helper.make_model(graph, producer_name='model')
checker.check_model(model, full_check=True)
inferred_shape = shape_inference.infer_shapes(model)
print(inferred_shape)
```
- Attach the ONNX model to the issue (where applicable)

### Notes

Any additional information

**Comment by jcwchen** [comment]
Hi @GuanLuo,
Thank you for identifying this issue. I can verify it -- ONNX shape inference for Transpose does not check whether any perm value is repeated: https://github.com/onnx/onnx/blob/master/onnx/defs/tensor/defs.cc#L1157, but NumPy and [ONNXRuntime](https://github.com/microsoft/onnxruntime/blob/0367e1f1c27ab097b4272f29e880843568c29e95/onnxruntime/core/providers/cpu/tensor/transpose.h#L55) do check. It should be improved and anyone interested in enhancing this check is welcome. Thanks!
